dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -509,6 +509,11 @@ private boolean isMemberAccessibleFrom(
       if (ClassDumper.classesInSamePackage(targetClass.getClassName(), sourceClassName)) {
         return true;
       }
+      if (Object.class.getName().equals(targetClass.getClassName())) {
```

Comment:
Does it also need `... && member.notPrviate()`?

Reply:
Line 508 checks whether this member is protected or not. I don't think a method can have protected and private.

Line 519 checks whether it's subclass or not. Now I think this condition should be in `ClassDumper.isClassSubClassOf`. Updating this.

Reply:
I changed my mind. Line 519 requires to load java.lang.Object. But the current code (c883b018fe42a26aa5165706d175dc8987f8f6f0) doesn't require it and it's obvious that all objects are subclass of java.lang.Object. So I keep the code as it is.

Reply:
Line 508 checks whether this member is protected or not.

Ah, I missed it.


> Also, should we ensure `targetClass` is `sourceClass` or a subclass of it?

Maybe I misunderstood the context, but what I mean above is this:
```java
  Object obj = new Object();
  Runnable runnable = new Runnable() {
    @Override
    public void run() {
      // sourceClass = runnable
      // targetClass = obj
      // runnable cannot call obj.clone() as it's protected
      obj.clone();  // doesn't compile
    }
  };
```
Or, is it that I'm out of the context?

Reply:
Thank you for the example. That's a new perspective to me. Unfortunately Linkage Checker cannot determine such cases. In the example below, they are both the same 
<img width="673" alt="Screen Shot 2021-08-12 at 15 30 16" src="https://user-images.githubusercontent.com/28604/129257367-f0f76c90-69e7-4c1d-97d7-f83b0a69ccc4.png">



As Linkage Checker deal with class files in JAR files, we can assume the code already passed the compilation.

Reply:
Cool, that's fair!

Reply:
Created an issue for this potential false negatives.

https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/2184